02week/ticTacToe.js

```diff
@@ -25,22 +27,134 @@ function printBoard() {
+    && board[2][1] === board[2][2])) {
+    return true;
+  }
 }
 
 function verticalWin() {
   // Your code here
+  // in row, column format these combinations of matching x's or o's will return a win
+  // 00 10 20
+  // 01 11 21
+  // 02 12 22
+  if ((board[0][0] === board[1][0] && board[1][0] === board[2][0]) || (board[0][1] === board[1][1] && board[1][1] === board[2][1]) || (board[0][2] === board[1][2]
+    && board[1][2] === board[2][2])) {
+    return true;
+  }
 }
 
 function diagonalWin() {
   // Your code here
+  // in row, column format these combinations of matching x's or o's will return a win
+  // 00 11 22
+  // 02 11 20
+  if ((board[0][0] === board[1][1] && board[1][1] === board[2][2]) || (board[0][2] === board[1][1] && board[1][1] === board[2][0])) {
+    return true;
+  }
 }
 
+// const winningCombinations = horizontalWin() || verticalWin() || diagonalWin();
+
+// winningCombinations = (row, column) => {
+//   [
+//     // horizontal wins
+//     [[0,0], [0,1], [0,2]], [[1,0], [1,1], [1,2]], [[2,0], [2,1], [2,2]],
+//     // verticals wins
+//     [[0,0], [1,0], [2,0]], [[0,1], [1,1], [2,1]], [[0,2], [1,2], [2,2]],
+//     // diagonal wins
+//     [[0,0], [1,1], [2,2]], [[0,2], [1,1], [2,0]];
+//   ];
+// }
+
 function checkForWin() {
   // Your code here
+  // if in any of the above combinations, there are either all x's or all o's then the respective player wins
+  // on each click, if horizontalWin or verticalWin or diagonalWin - gameover and ${} player wins. If no win, then alternate player.
+
+  // for (var i = 0; i < winningCombinations.length; i++) {
+  //   let i, j
+  //   let count = 0;
+  //   for (let j = 0; j < winningCombinations[i].length; j++) {
+  //   if (board[winningCombinations[i][j]] === playerTurn) {
+  //       count++;
+  //   }if (count === 3) {
+  //       return true;
+  //   }
+  // }
+  // return false;
+  // }
+
+  if (horizontalWin() || verticalWin() || diagonalWin()) {
+    console.log(`Player ${playerTurn} Wins!`);
+  } else if (turnCount === 9) {
+    console.log("It's a tie!")
+  } else {
+    return false;
+  }
 }
 
 function ticTacToe(row, column) {
   // Your code here
+  // the playerturn will start at x but will alternate between xs and os onclicks, and for each click it will checkForWin.
+    // if win, game winner will be announced and board will reset.
+  // call printBoard after every turn (push playerTurn to array
+
+  
+  // if (checkForWin() = false) {
+    if (playerTurn === 'X') {
+      playerTurn = 'O';
+    } else {
+      playerTurn = 'X';
+    }
+  // }
+
+  // board[row][column] = playerTurn;
+  // if (moveCount > 0){
+  // checkForWin();
+  // }
+  // moveCount ++;
+  // playerTurn = (playerTurn === "X" ? "O" : "X");
+  // }
+
+  if (board[row][column] === ' ') {
+    board[row].splice(column, 1, playerTurn);
+  }
+
+
+
+  // const validValue = (myIndex) => {
```

Comment:
this should be un commented and moved to an outside function. Your logic is spot on, check for valid move, then the win, then switch the player.